### PR TITLE
Add endpoints for getting snapshot information

### PR DESCRIPTION
**Pull request by tenyo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -190,6 +190,97 @@ PUT http://127.0.0.1:3000/v1/rds/{account}/myaurora/power
 }
 ```
 
+### Getting a list of snapshots for a database/cluster
+
+This will return list of snapshots (with details) for the specified database in either `DBClusterSnapshots` or `DBSnapshots`, depending if it's a cluster or an instance.
+It will also set an `X-Items` header containing the total number of snapshots in the list.
```
Comment (fishnix) [inline README.md:196]: `X-Items` amazing!

File: actions/snapshots.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+	if clusterSnapshotsOutput, err = rdsClient.Service.DescribeDBClusterSnapshotsWithContext(c, &rds.DescribeDBClusterSnapshotsInput{DBClusterIdentifier: aws.String(c.Param("db"))}); err != nil {
+		return handleError(c, err)
+	}
+
+	if instanceSnapshotsOutput, err = rdsClient.Service.DescribeDBSnapshotsWithContext(c, &rds.DescribeDBSnapshotsInput{DBInstanceIdentifier: aws.String(c.Param("db"))}); err != nil {
+		return handleError(c, err)
+	}
+
+	if clusterSnapshotsOutput.DBClusterSnapshots == nil && instanceSnapshotsOutput.DBSnapshots == nil {
```
Comment (fishnix) [inline actions/snapshots.go]: you're probably better off using `len()` here, those values may be set and empty

File: actions/snapshots.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	}
+
+	log.Printf("getting snapshots for %s", c.Param("db"))
+
+	var clusterSnapshotsOutput *rds.DescribeDBClusterSnapshotsOutput
+	var instanceSnapshotsOutput *rds.DescribeDBSnapshotsOutput
+	var err error
+	var items int
+
+	if clusterSnapshotsOutput, err = rdsClient.Service.DescribeDBClusterSnapshotsWithContext(c, &rds.DescribeDBClusterSnapshotsInput{DBClusterIdentifier: aws.String(c.Param("db"))}); err != nil {
```
Comment (fishnix) [inline actions/snapshots.go]: holy long line 😆 

tbh in these cases, its better to not use the inline `if`.  then you will get rid of the `var` declarations above too - this ain't C yo....

File: actions/snapshots.go [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+	var clusterSnapshot *rds.DBClusterSnapshot
+	var instanceSnapshot *rds.DBSnapshot
+
+	if clusterSnapshotsOutput, err = rdsClient.Service.DescribeDBClusterSnapshotsWithContext(c, &rds.DescribeDBClusterSnapshotsInput{DBClusterSnapshotIdentifier: aws.String(c.Param("snap"))}); err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() != rds.ErrCodeDBClusterSnapshotNotFoundFault {
+				return c.Error(400, aerr)
+			}
+		}
+	} else if len(clusterSnapshotsOutput.DBClusterSnapshots) == 1 {
```
Comment (fishnix) [inline actions/snapshots.go]: what if its `>1`?  will the next section return an error?

File: actions/snapshots.go [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+			if aerr.Code() != rds.ErrCodeDBSnapshotNotFoundFault {
+				return c.Error(400, aerr)
+			}
+		}
+	} else if len(instanceSnapshotsOutput.DBSnapshots) == 1 {
+		isInstanceSnapshot = true
+		instanceSnapshot = instanceSnapshotsOutput.DBSnapshots[0]
+	}
+
+	if !isClusterSnapshot && !isInstanceSnapshot {
```
Comment (fishnix) [inline actions/snapshots.go]: here you can probably just check if `instanceSnapshot` and  `clusterSnapshot` are nil and get rid of the bools

**Review by fishnix — COMMENTED** [review]
Just a few minor things, otherwise looking good!

File: actions/snapshots.go (hunk 5 above)
Comment (tenyo) [reply actions/snapshots.go]: how did i miss that 🤦

**Comment by tenyo** [comment]
Good comments, much better now!

File: actions/snapshots.go (hunk 5 above)
Comment (fishnix) [reply actions/snapshots.go]: thats what PRs are for 👍

**Review by fishnix — APPROVED** [review]
looks good! 👍

**Pull request merged**